### ISSUE TITLE
Is it possible to detect untrained object as unknown object using yolov5?

### ISSUE THREAD

**Issue by Su-Bi-su** [ultralytics/yolov5#6241]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have a problem case where an image should have fixed objects in any positions within an image. 
The know objects are fixed. 
I want to develop a solution where I want to detect if any object that is not supposed to be in the image is present. 

I can easily detect all the known objects. But is it possible to detect an object that was not trained but is present in the image using yolov5? I want to detect this untrained object as an "unknown" object.

**Comment by glenn-jocher** [comment]
@Su-Bi-su YOLOv5 is an object detector, you can train it to detect any objects or areas in images that you can label.

**Comment by Zephyr69** [comment]
It's possible, but you need to modify the training loop, design a new loss, and possibly modify model arch.

**Comment by glenn-jocher** [comment]
@Zephyr69 thanks for your input! If you have the necessary expertise, you could modify the training process to accomplish this. Good luck with your implementation!

**Comment by choudharyprince890** [comment]
> It's possible, but you need to modify the training loop, design a new loss, and possibly modify model arch.

can you please elaborate

**Comment by glenn-jocher** [comment]
@choudharyprince890 To detect unknown objects, you could modify the training process by expanding the training dataset to include a new "unknown" class, and adjusting the loss function and model architecture accordingly. This involves some advanced customization and coding. For more details, you can refer to the Ultralytics YOLOv5 documentation at https://docs.ultralytics.com/yolov5/. Good luck!
